### PR TITLE
Enable frozen string literals throughout

### PR DESCRIPTION
**Pull request by tas50**
This doesn't actually get us any memory allocation improvements in this project, but it's going to be the default eventually so it gets us ready for that. There was one "" that needed to become a String.new instead to avoid failures.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,10 +1,11 @@
+# frozen_string_literal: true
 source "https://rubygems.org"
 
 gemspec
 
 # NOTE: do not submit PRs to add pry as a dep, add to your Gemfile.local
 group :development do
-  gem "chefstyle", git: "https://github.com/chef/chefstyle.git", branch: "master"
+  gem "chefstyle", "= 1.1.3"
```
Comment (tas50) [inline Gemfile:8]: Using chefstyle from git just resulted in constant errors locally and the inability to use dependabot to bump style on our own terms. This way it'll open PRs to bump the version and we can evaluate any fixes that need to happen instead of breaking community PRs potentially due to unrelated rubocop changes.

File: lib/ohai/loader.rb [hunk 2]
```diff
@@ -98,7 +99,7 @@ def load_plugin(plugin_path)
     # @return [Object] class object for the ohai plugin defined in the file
     def load_plugin_class(plugin_path)
       # Read the contents of the plugin to understand if it's a V6 or V7 plugin.
-      contents = ""
+      contents = String.new
       begin
         logger.trace("Loading plugin at #{plugin_path}")
         contents << IO.read(plugin_path)
```
Comment (phiggins) [inline lib/ohai/loader.rb:105]:
```suggestion
      contents = begin
        logger.trace("Loading plugin at #{plugin_path}")
        IO.read(plugin_path)
```

I mean, if we're really about saving memory with this PR...

Comment (tas50) [reply lib/ohai/loader.rb:105]: Wouldn't we want the trace outside the begin there?

File: lib/ohai/plugins/aix/virtualization.rb [hunk 3]
```diff
@@ -91,7 +91,7 @@
                       }
               wpars[wpar_name][title][key] = value
             when "security settings"
-              privileges ||= ""
+              privileges = String.new
```
Comment (jaymzh) [inline lib/ohai/plugins/aix/virtualization.rb:94]: to be clear, these aren't the same, you used to preserve any setting of priveleges, now you're deleting it.

File: lib/ohai/plugins/aix/virtualization.rb [hunk 4]
```diff
@@ -140,7 +140,8 @@
         ]
 
         top_level.each do |attribute|
-          evalstr = "wpars['#{wpar_name}']"
+          evalstr = String.new
+          evalstr << "wpars['#{wpar_name}']"
```
Comment (jaymzh) [inline lib/ohai/plugins/aix/virtualization.rb:144]: why is this better. than `evalstr = "wpars['#{wpar_name}']".freeze` ?

File: lib/ohai/plugins/ruby.rb [hunk 5]
```diff
@@ -47,13 +46,13 @@ def run_ruby(command)
...
-    values.each_key do |v|
-      env_string << "#{v}=\#{#{values[v]}},"
+    env_string = []
+    values.each_pair do |k,v|
+      env_string << "#{k}=\#{#{v}}"
     end
 
     # Query the system ruby
-    result = run_ruby "puts %Q(#{env_string})"
+    result = run_ruby "puts %Q(#{env_string.join(',')})"
```
Comment (jaymzh) [inline lib/ohai/plugins/ruby.rb:55]: <3

File: lib/ohai/loader.rb (hunk 2 above)
Comment (phiggins) [reply lib/ohai/loader.rb:105]: Yeah that probably makes more sense 👍

File: lib/ohai/plugins/ruby.rb [hunk 6]
```diff
@@ -46,13 +46,13 @@ def run_ruby(command)
     }
 
     # Create a query string from above hash
-    env_string = ""
-    values.each_key do |v|
-      env_string << "#{v}=\#{#{values[v]}},"
+    env_string = []
+    values.each_pair do |k,v|
+      env_string << "#{k}=\#{#{v}}"
     end
```
Comment (phiggins) [inline lib/ohai/plugins/ruby.rb:52]:
```suggestion
    env_string = values.map do |k,v|
      "#{k}=\#{#{v}}"
    end
```

File: spec/unit/plugins/passwd_spec.rb [hunk 7]
```diff
@@ -55,7 +56,7 @@
   if "".respond_to?(:force_encoding)
```
Comment (phiggins) [inline spec/unit/plugins/passwd_spec.rb:56]: `.force_encoding` was added in ruby 2.0 (?) so this check is probably safe to remove.

File: lib/ohai/plugins/passwd.rb [hunk 8]
```diff
@@ -9,7 +10,7 @@
   # @return [String]
   #
   def fix_encoding(str)
-    str.force_encoding(Encoding.default_external) if str.respond_to?(:force_encoding)
+    str.dup.force_encoding(Encoding.default_external) if str.respond_to?(:force_encoding)
```
Comment (phiggins) [inline lib/ohai/plugins/passwd.rb:13]:
```suggestion
    str.dup.force_encoding(Encoding.default_external)
```

`.force_encoding` was added a long time ago, probably safe to remove.

**Comment by tas50** [comment]
I think I'm going to hit reset on this and only set frozen where we're not mutating for now
